app/web/src/helpers/hooks/useIdeState.ts

```diff
@@ -146,103 +147,134 @@ export const initialState: State = {
+  console.log('reducing')
+  switch (type) {
+    case 'initIde':
+      return {
+        ...state,
+        code:
+          payload.code ||
+          // localStorage.getItem(makeCodeStoreKey(payload.cadPackage)) ||
+          initCodeMap[payload.cadPackage] ||
+          '',
+        ideType: payload.cadPackage,
+      }
+    case 'updateCode':
+      return { ...state, code: payload }
+    case 'healthyRender':
+      const customizerParams: CadhubParams[] = payload?.customizerParams?.length
+        ? payload.customizerParams
+        : state.customizerParams
+      const currentParameters = {}
+      customizerParams.forEach((param) => {
+        currentParameters[param.name] =
+          typeof state?.currentParameters?.[param.name] !== 'undefined'
+            ? state?.currentParameters?.[param.name]
+            : param.initial
+      })
+      return {
+        ...state,
+        objectData: {
+          ...state.objectData,
+          type: payload.objectData?.type,
+          data: payload.objectData?.data,
+        },
+        customizerParams,
+        currentParameters,
+        consoleMessages: payload.message
+          ? [...state.consoleMessages, payload.message]
+          : payload.message,
+        isLoading: false,
+      }
+    case 'errorRender':
+      return {
+        ...state,
+        consoleMessages: payload.message
+          ? [...state.consoleMessages, payload.message]
+          : payload.message,
+        isLoading: false,
+      }
+    case 'setCurrentCustomizerParams':
+      if (!Object.keys(payload || {}).length) return state
+      return {
+        ...state,
+        currentParameters: payload,
+      }
+    case 'setLayout':
+      return {
+        ...state,
+        layout: payload.message,
+      }
+    case 'updateCamera':
+      return {
+        ...state,
+        camera: payload.camera,
+      }
+    case 'updateViewerSize':
+      return {
+        ...state,
+        viewerSize: payload.viewerSize,
+      }
+    case 'setLoading':
+      return {
+        ...state,
+        isLoading: true,
+      }
+    case 'resetLoading':
+      return {
+        ...state,
+        isLoading: false,
+      }
+    case 'resetLayout':
+      return {
+        ...state,
+        layout: initialLayout,
+      }
+    case 'setThreeInstance':
+      return {
+        ...state,
+        threeInstance: payload,
+      }
+    case 'settingsButtonClicked':
+      const isReClick =
+        state.sideTray.length &&
+        state.sideTray.length === payload.length &&
+        state.sideTray.every((original, index) => original === payload[index])
+
+      const payloadInOriginal =
+        payload.length && state.sideTray.indexOf(payload[0])
+      const isAncestorClick =
+        state.sideTray.length &&
+        state.sideTray.length > payload.length &&
+        payloadInOriginal >= 0 &&
+        payload.every(
+          (incoming, i) => incoming === state.sideTray[i + payloadInOriginal]
+        )
```

Comment:
This is how I implemented clicks on the main gear icon closing the whole menu. Does this work for you @Irev-Dev?